core/src/main/scala/org/apache/spark/storage/StorageLevel.scala

```diff
@@ -86,7 +83,7 @@ class StorageLevel private(
       false
   }
 
-  def isValid: Boolean = (useMemory || useDisk || useOffHeap) && (replication > 0)
+  def isValid: Boolean = (useMemory || useDisk) && (replication > 0)
```

Comment:
I always thought that `isValid` was somewhat confusingly-named. From its current usage, `isValid` means "if we interpret this StorageLevel as a block's effective StorageLevel (i.e. a descriptor of where it's actually stored) then the block exists in at least one block store." Under that interpretation, I believe that this definition of `isValid` is correct.